plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/TimestampDecoder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.decoders;
+
+import io.trino.plugin.pinot.conversion.PinotTimestamps;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.function.Supplier;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static java.lang.String.format;
+import static java.time.ZoneOffset.UTC;
+
+public class TimestampDecoder
+        implements Decoder
+{
+    @Override
+    public void decode(Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else {
+            LocalDateTime timestamp;
+            if (value instanceof String) {
```

Comment:
How is timestamp internally represented ? As STRING or as LONG ?

Reply:
Pinot broker query gives String and Pinot server query results give numeric value.

Reply:
But the decoder is used by `PinotBrokerPageSource` so can we handle only `String` here ?

Reply:
done

Reply:
Again checked, seems that some functions may return numeric values, so still need to do the type check:
```
Suppressed: java.lang.Exception: SQL: SELECT COUNT(*),  MIN(int_col), MAX(int_col),  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col),  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col),  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col),  MIN(timestamp_col), MAX(timestamp_col)  FROM alltypes
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:483)
		... 18 more
```